src/server/app.py

```diff
@@ -4,37 +4,77 @@
 api_email = 'davethecust'
 api_key = 'bMSeYjdu2GA5CtC4Dc3dPz8l6C1VgS4x5bpnV4cv'
 tournament_url = '42k5wcre'
+
+
+challonge.set_credentials(api_email, api_key)
 tournament_id = challonge.tournaments.show(tournament_url)['id']
-players = []
+players = {p['id']: p['name']
+           for p in challonge.participants.index(tournament_id)}
 matches = []
+
+
 app = Flask('auTO')
 
+
 # method takes in a state from the tournament and updates challonge
 # state_dump should be passed the following data via http PUT
 #
 # {
-#   'winner_id': int,
-#   'scores_csv': string - "3-1"
+#   'winner': string,
+#   'scores_csv': string ex, "3-1"
 # }
+#
+# i'm making the assumption the UI will have text boxes for
+# 'tag' and 'games won', 'games lost'
+# where for example, tag = hamroctopus, games won = 3, games lost = 1
+# in the front end we put together a json object that is similar to the above
+#
+# challonge api weird thing, if player two wins, we need to reverse the games won
+# when updating the match, you have to do things centric to player one
+# this explanation probably doesn't make sense,
+# please see the challonge api reference -
+#   http://api.challonge.com/v1/documents/matches/update
```

Comment:
oh yeah, i see. should my proposed fields on the ui be changed? can you think of a scheme that makes more sense for both users and devs? else we just deal with shitty logical hoops